test-data/unit/check-overloading.test

```diff
@@ -6545,3 +6545,21 @@ class Snafu(object):
 reveal_type(Snafu().snafu('123'))  # N: Revealed type is "builtins.str"
 reveal_type(Snafu.snafu('123'))  # N: Revealed type is "builtins.str"
 [builtins fixtures/staticmethod.pyi]
+
+[case testOverloadedFunctionWithTypevarMissing]
+import typing
+
+class A: ...
+
+T = typing.TypeVar("T", bound=A)
```

Comment:
I should investigate why current mypy is a-ok without the `bound=A` here, but errors when it is. That's strange and probably a sign of incorrect special casing: something I hate!